### PR TITLE
Validate max size for images

### PR DESCRIPTION
**Pull request by gcampo88**
#447 add max-size messaging + validations for series and story images.

### CODE REVIEW

File: src/app/shared/image/image-upload.component.ts [hunk 1]
```diff
@@ -33,6 +38,8 @@ export class ImageUploadComponent implements DoCheck {
   @Input() purpose: string;
   @Input() minWidth = 144;
   @Input() minHeight = 144;
+  @Input() maxWidth = 144;
+  @Input() maxHeight = 144;
```
Comment (sandikbarr) [inline src/app/shared/image/image-upload.component.ts]: The series thumbnail image isn't working. Can we maybe not have a default max or something? Or maybe just in the validation below check for either suggestSize or min/max?
screen shot 2017-09-07 at 5 52 56 pm

Comment (gcampo88) [reply src/app/shared/image/image-upload.component.ts]: oooh, good catch!

**Comment by gcampo88** [comment]
good catch @sandikbarr. back to you!

File: src/app/shared/image/image-upload.component.ts [hunk 2]
```diff
@@ -126,6 +117,24 @@ export class ImageUploadComponent implements DoCheck {
...
+    if (!this.suggestSize) {
+      if (width < this.minWidth || height < this.minHeight) {
+        this.imgError = `The image provided is only ${width} x ${height} px
+                         but should be at least ${this.minWidth} x ${this.minHeight} px.`;
+        return false;
+      } else if ((this.maxWidth && width > this.maxWidth) || (this.maxHeight && height > this.maxHeight)) {
+        this.imgError = `The image provided is ${width} x ${height} px
+                         but should not exceed ${this.maxWidth} x ${this.maxHeight} px.`;
+        return false;
+      } else if (this.square && width !== height) {
```
Comment (sandikbarr) [inline src/app/shared/image/image-upload.component.ts]: I wondered if we wanted to enforce square without enforcing min or max, but it looks like we're not doing that, so 👍

**Pull request merged**